### ISSUE TITLE
Retrieval permission to deals

### ISSUE THREAD

**Issue by kernelogic** [filecoin-project/FIPs#227]
## Summary

(OP @Da-you. Proofread by @Fei Yan)

Add retrieval permission to deals, allowing only certain clients to retrieve a deal.

## Motivation

Currently Filecoin protocol allows anyone with the data CID to retrieve a deal. 

In a real word scenario, privacy and security are important for data storage, and it is relatively limiting if your stored data can be freely retrieved by others, although we can encrypt that data, but preventing others from retrieving it on the protocol layer can provide higher security and usability.

## Design

Client side:  When proposing a deal, you can specify one or more client addresses that are allowed to retrieve that data.
SP side : When a client retrieves a deal, first check to see if the client is in the list of clients that can be retrieved. If the list is empty, then it is open to everyone.

## Use Cases

Personal or business surveillance video requires a high level of security and privacy, they may not retrieve data often, but when they want to, they want to be the only ones who can.

## Consideration

Make updates to the allowed list after deal is on chain.

**Comment by placer14** [comment]
Wouldn't a private encryption scheme over the data better serve this need without adding this behavior to FIL protocol? IMHO, permissions/ACLs don't feel like they belong at this layer.

Can you explain why this might be preferred over my suggestion to use encryption?

**Comment by anorth** [comment]
In line with https://github.com/filecoin-project/FIPs/discussions/215, this should be moved to the discussion forum. There is not yet any concrete proposed protocol change to evaluate as a FIP.

**Comment by dayou5168** [comment]
In my impression, if you use an existing storage service such as aws s3, aliyun, qiniuyun, your files will not be accessed by anyone else whether they are encrypted or not, please let me know if my perception is wrong

**Comment by RobQuistNL** [comment]
I don't think it makes sense from a security perspective. It does however make sense from a bandwidth / DDOS / data-hosting-contract perspective :)

**Comment by jennijuju** [comment]
> In my impression, if you use an existing storage service such as aws s3, aliyun, qiniuyun, your files will not be accessed by anyone else whether they are encrypted or not, please let me know if my perception is wrong

Well the service platform has direct access to the data.
